src/components/select/UISelect.vue

```diff
@@ -331,34 +341,46 @@ export default defineComponent({
 
             selected.value = props.multi ? [] : null;
         };
+
         const isEqual = (previous: Option, next: Option): boolean => {
+            // Compare as strings options (checking type of previous/next to satisfy TS).
+            if (stringOptions.value || typeof previous === 'string' || typeof next === 'string') {
+                const previousString: string = typeof previous === 'string' ? previous : previous[props.optionLabel];
+                const nextString: string = typeof next === 'string' ? next : next[props.optionLabel];
+
+                return previousString.toLowerCase() === nextString.toLowerCase();
```

Comment:
I don't thinkwe should lowercase, if it's a different value, liets assume it is intentionally.
Especially that this would break if the optionKey was use like in the above comment as on number there is no such method.

Reply:
I can remove this for now but I expect we'll be revisiting this to improve the case handling after a bit of real-world use.